// sslcommerz/pop-up.js
import tingle from 'tingle.js';

export const loadEasyCheckout = (body, isLive = false, isMobile = false) => {
    const url = body.url;
    const logo = body.logo;
    window.addEventListener('message', (e) => {
        // console.log('WE HAVE RECEIVED ALIEN MESSAGE', e);
        if (e.data !== undefined) {
            let data = {};
            try {
                data = JSON.parse(e.data);
            } catch (e) {
            }
            // console.log(data);
            if (data && data.status !== undefined) {
                if (data.status === 'success') {
                    if (data.url !== undefined) {
                        if (data.url.indexOf('http') !== -1) {
                            // URL TO Redirect
                            // popup.close();
                            window.location.href = data.url;
                        } else {
                            data = JSON.parse(data.data);
                            document.write(data.data.url);
                        }
                    }

                    if (data.type !== undefined) {
                        if (data.type === 'redirect' || data.type === 'gw_redirect') {
                            // modal.close();
                            // popup.close();
                            // closeSSLPopup();
                            window.location.href = data.url;
                        }
                    }

                    if (data.type !== undefined) {
                        if (data.type === 'otp') {
                            try {
                                modal.close();
                                if (isLive) {
                                    document.open();
                                    document.write(data.data);
                                } else {
                                    data = JSON.parse(data.data);
                                    document.write(data.data.url);
                                }
                            } catch (e) {
                                document.write(data.data);
                            }
                            document.close();
                        } else {
                            if (isMobile) {
                                window.location.href = url;
                            }
                        }
                    }

                    if (data.type !== undefined) {
                        if (data.type === 'resize') {

                            if (data.height !== undefined) {
                                if (data.height > 100) {
                                    if (isMobile) {
                                        // console.log(this.isMobile);
                                        /*iframe.height = window.outerHeight - 100;*/
                                        iframe.style.minHeight = window.outerHeight - 100 + 'px';
                                        /* console.log('iframe height'+iframe.height);*/
                                    } else {
                                        iframe.height = data.height;
                                    }

                                }

                            }

                        }
                    }
                }
            }
        }
    }, false);

    const modal = new tingle.modal({
        footer: true,
        stickyFooter: false,
        closeMethods: ['button'],
        closeLabel: 'Close',
        cssClass: ['custom-class-1', 'custom-class-2'],
        onOpen: () => {
            // console.log('modal open');
        },
        onClose: () => {
            // console.log('modal closed');
            // closeSSLPopup();
        },
        beforeClose: () => {
            return true; // close the modal

        }
    });

    const mainDiv = document.createElement('div');
    const innerDiv = document.createElement('div');
    const inerInnerDiv = document.createElement('div');
    innerDiv.className = 'client_logo';
    inerInnerDiv.className = 'client_logo_inner';
    const iframeDiv = document.createElement('div');
    iframeDiv.setAttribute('class', 'iframeWrap');
    const iframe = document.createElement('iframe');
    iframe.src = url + '?full=1';
    iframe.name = 'frame';
    iframe.setAttribute('allowFullScreen', '');
    iframe.allowFullScreen = true;


    iframe.width = 340;
    iframe.setAttribute('frameborder', '0');
    iframe.scrolling = 'no';
    iframe.allowtransparency = true;
    iframe.seamless = true;
    iframeDiv.appendChild(iframe);
    const oImg = document.createElement('img');
    oImg.setAttribute('src', logo);
    oImg.setAttribute('alt', 'Logo');

    inerInnerDiv.appendChild(oImg);
    innerDiv.appendChild(inerInnerDiv);
    mainDiv.appendChild(innerDiv);
    mainDiv.appendChild(iframeDiv);
    modal.setContent(mainDiv);
    // add another button
    modal.addFooterBtn('&times;', 'tingle-modal__close', () => {
        // here goes some logic
        modal.close();
    });
    modal.open();
}
